src/gui/instrument/EnvelopeGraph.cpp

```diff
@@ -54,102 +56,208 @@ void EnvelopeGraph::modelChanged()
 
 void EnvelopeGraph::mousePressEvent(QMouseEvent* me)
 {
-	if(me->button() == Qt::LeftButton)
+	if (me->button() == Qt::LeftButton) { toggleAmountModel(); }
+}
+
+void EnvelopeGraph::contextMenuEvent(QContextMenuEvent* event)
+{
+	QMenu menu(this);
+	QMenu* scalingMenu = menu.addMenu(tr("Scaling"));
+	scalingMenu->setToolTipsVisible(true);
+
+	auto switchTo = [&](ScalingMode scaling)
 	{
-		toggleAmountModel();
-	}
+		if (m_scaling != scaling)
+		{
+			m_scaling = scaling;
+			update();
+		}
+	};
+
+	auto addScalingEntry = [scalingMenu, &switchTo, this](const QString& text, const QString& toolTip, ScalingMode scaling)
+	{
+		QAction* action = scalingMenu->addAction(text, [&switchTo, scaling]() { switchTo(scaling); });
+		action->setCheckable(true);
+		action->setChecked(m_scaling == scaling);
+		action->setToolTip(toolTip);
+	};
+	
+	addScalingEntry(tr("Dynamic"), tr("Uses absolute spacings but switches to relative spacing if it's running out of space"), ScalingMode::Dynamic);
+	addScalingEntry(tr("Absolute"), tr("Provides enough potential space for each segment but does not scale"), ScalingMode::Absolute);
+	addScalingEntry(tr("Relative"), tr("Always uses all of the available space to display the envelope graph"), ScalingMode::Relative);
```

Comment:
Kind of congested, may be easier to read when split apart.
```suggestion
	addScalingEntry(
		tr("Dynamic"),
		tr("Uses absolute spacings but switches to relative spacing if it's running out of space"),
		ScalingMode::Dynamic
	);
	addScalingEntry(
		tr("Absolute"),
		tr("Provides enough potential space for each segment but does not scale"),
		ScalingMode::Absolute
	);
	addScalingEntry(
		tr("Relative"),
		tr("Always uses all of the available space to display the envelope graph"),
		ScalingMode::Relative
	);
```

Reply:
Adjusted with commit af4672b2094.